shell/browser/native_window_views.cc

```diff
@@ -668,6 +668,22 @@ bool NativeWindowViews::IsResizable() {
   return CanResize();
 }
 
+void NativeWindowViews::SetAspectRatio(double aspect_ratio,
+                                       const gfx::Size& extra_size) {
+#if defined(OS_LINUX)
+  NativeWindow::SetAspectRatio(aspect_ratio, extra_size);
+  gfx::SizeF aspect(aspect_ratio, 1.0);
+  // widget SetAspectRatio() won't work with numbers too small
+  aspect.Scale(100);
+
+  if (aspect_ratio > 0) {
+    widget()->SetAspectRatio(aspect);
+  } else {
+    // reset aspect ratio somehow
```

Comment:
There doesn't seem to be any explicit way of doing this in the existing code, at least for Linux. It looks like the low-level implementation is at `DesktopWindowTreeHostX11::SetAspectRatio` where the X11 `PAspect` flag gets turned on for`XSizeHints` ... and there's no other place where `PAspect` is used explicitly again.

If we wanted to add support for unsetting, I think we'd have to add it ourselves. For example in the Linux code we'd need something like this:

```diff
--- a/ui/views/widget/desktop_aura/desktop_window_tree_host_x11.cc
+++ b/ui/views/widget/desktop_aura/desktop_window_tree_host_x11.cc
@@ -1129,9 +1129,13 @@ void DesktopWindowTreeHostX11::SetAspectRatio(const gfx::SizeF& aspect_ratio) {
   long supplied_return;
 
   XGetWMNormalHints(xdisplay_, xwindow_, &size_hints, &supplied_return);
-  size_hints.flags |= PAspect;
-  size_hints.min_aspect.x = size_hints.max_aspect.x = aspect_ratio.width();
-  size_hints.min_aspect.y = size_hints.max_aspect.y = aspect_ratio.height();
+  if (aspect_ratio.IsEmpty()) {
+    size_hints.flags &= ~PAspect;
+  } else {
+    size_hints.flags |= PAspect;
+    size_hints.min_aspect.x = size_hints.max_aspect.x = aspect_ratio.width();
+    size_hints.min_aspect.y = size_hints.max_aspect.y = aspect_ratio.height();
+  }
   XSetWMNormalHints(xdisplay_, xwindow_, &size_hints);
 }
```

Maybe upstream would want a patch like this, but I'm not sure -- they probably don't have a use case for it?

Reply:
I found the relevant bit of code in `x11_window.cc`, but implemented the same logic!